tsds-grafana-handler/tsds_grafana_handler.py

```diff
@@ -152,8 +152,22 @@ def convert(innerValue):
         return [[element[0],element[1]*1000] for element in IV]
 
 #Replace $START, $END, and $quantify variables in query with its corresponding value
-def replaceQuery(query,start_time,end_time, aggValue):
-        replace = {"$START":start_time,"$END":end_time, "$quantify":str(aggValue), "{":" ", "}":" "}
+def replaceQuery(query,start_time,end_time, duration, buckets):
```

Comment:
This function needs to consider one additional thing. The workflow should be:

Calculate the auto-calc bucket size based on the timerange compared to the max data points (grafana sends this in the request). This should still factor in the cases where if the timerange > certain thresholds, it gets raised to 3600 and 86400 approriately.

Then take the max of that autocalc or the requested size. This means a requested 3600 will get upgraded to 86400 at longer time ranges. It also means that a requested 3600 could be upgraded to (86400 * 7) if looking at a REALLY long timerange for example.

Reply:
Forgot to mention - it looks like this commit condenses the dupe code doing this in a few places which is good. Looking at one of those place it looks like maxdatapoints was hard coded to 1, not sure why that was but it might be that for some things such as template vars we don't have that set, which is okay. There is another location starting at line 338 that was doing this correctly.